pool.go

```diff
@@ -282,42 +296,50 @@ func (p *pool) replaceNodeToHaveWeight(nm *Member) {
 	}
 }
 
-func (p *pool) fetchAndUpdate(ctx context.Context, node string, c cid.Cid, attempt int) (blk blocks.Block, err error) {
+func (p *pool) fetchAndUpdate(ctx context.Context, node string, c cid.Cid, attempt int, transientErrs map[string]struct{}) (blk blocks.Block, err error) {
 	blk, err = p.doFetch(ctx, node, c, attempt)
 	if err != nil {
 		goLogger.Debugw("fetch attempt failed", "from", node, "attempt", attempt, "of", c, "error", err)
 	}
 
-	var idx int
-	var nm *Member
-
 	if err == nil {
-		// Saturn fetch worked, we should try upvoting.
-		p.lk.RLock()
-		idx, nm = p.updateWeightUnlocked(node, false)
-		p.lk.RUnlock()
-
-		if idx != -1 && nm != nil && p.endpoints[idx].url == nm.url {
-			p.replaceNodeToHaveWeight(nm)
-		}
-
+		p.upvote(node)
 		// Saturn fetch worked, we return the block.
 		return
 	}
 
+	// If this is a NOT found or Timeout error, park the downvoting for now and see if other members are able to give us this content.
+	if errors.Is(err, ErrContentNotFound) || errors.Is(err, ErrSaturnTimeout) {
+		transientErrs[node] = struct{}{}
+		return
+	}
+
 	// Saturn fetch failed, we downvote the failing member.
+	p.downvote(node)
+	return
+}
+
+func (p *pool) upvote(node string) {
 	p.lk.RLock()
-	idx, nm = p.updateWeightUnlocked(node, true)
+	idx, nm := p.updateWeightUnlocked(node, false)
+	p.lk.RUnlock()
+
+	if idx != -1 && nm != nil {
+		p.replaceNodeToHaveWeight(nm)
+	}
+}
+
+func (p *pool) downvote(node string) {
+	p.lk.RLock()
+	idx, nm := p.updateWeightUnlocked(node, true)
 	p.lk.RUnlock()
 
 	// we weren't able to downvote the node.
 	if idx == -1 || nm == nil {
 		return
 	}
 
-	// we need to take the lock as we're updating the list of pool endpoint members below.
 	p.replaceNodeToHaveWeight(nm)
-	return
 }
```

Comment:
can `upvote` and `downvote` be one method with `direction bool` since it seems like they basically just differ in the `updateWeightUnlocked` bool